lib/git-version-bump.rb

```diff
@@ -224,6 +224,32 @@ def self.run_command(cmd, desc)
 	end
 	private_class_method :run_command
 
+        # Execute a command, specified as an array but retain ability to run the editor for notes.
+        #
+        # On success, control is returned.
+        # On error, a `CommandFailure` exception is raised.
+        #
+        def self.run_command_retain_tty(cmd, desc)
```

Comment:
Given how much of this method's body is shared with `run_command`, I'm inclined to add a kwarg (with a default) to `run_command` to specify "don't capture the command output", rather than duplicating the method's code.  It's a close-run thing, because it's complexity either way -- a zillion kwargs is at least as much of a mess as near-duplicate methods, but in this case I *think* the kwarg is better.

Reply:
I agree with that, was trying to "when in rome..." :) But your call.  Happy to help in any way.  I really dislike copy / paste.  The other thing I was thinking was maybe it should return the status.exitcode, but we weren't checking the return code so I didn't do that.  But it would be cleaner to return something that might be helpful, in my opinion.

Reply:
I decided to raise an exception when a command fails, rather than return the exit code, because a failing command is (usually) exceptional.

I think what feels best is to:

1. change the `run_command` method to have the signature of `run_command(cmd, desc, capture_output: true)';
2. adjust the method code to match;
3. when output isn't being captured:
  1. the `out` arg to the exception can be `(output not captured)`, as you've done in your commit;
  2. the return value is `nil`;
  3. `spawn` is called rather than `capture2e`.

Reply:
I'll give it a whirl.

The pros of "run_command2" are I wouldn't have spent so much time thinking about a method that we throw out anyways. :)   At first to see if it worked I just hard coded system() instead of the run_command call. :)  Made me feel too dirty to even submit.

Reply:
I'll give it a whirl.

Reply:
Me and GUIs are not usually friends.  I think it says I've offered the commit.  Can you validate?  The only thing I was thinking as I wrote it is maybe we should be checking for true, false and else (unsupported value).  But that didn't seem to be consistent with what you wanted.  How's this look?